botorch/posteriors/multitask.py

```diff
@@ -286,3 +287,38 @@ def _draw_from_base_covar(
         res = covar_root.matmul(base_samples)
 
         return res.squeeze(-1)
+
+
+def _permute_solve(A: LinearOperator, b: LinearOperator) -> LinearOperator:
```

Comment:
Would you mind writing a unit test for this method to make sure this doesn't break e.g. with updates to `linear_operator`? Basically just add a `TestPermuteSolve` test case to https://github.com/pytorch/botorch/blob/main/test/posteriors/test_multitask.py that checks that the solution to this is correct (either by comparing it against unpermuted solves or by comparing it against a few known solutions on small examples.